src/main/groovy/lang/MetaClassImpl.java

```diff
@@ -2314,7 +2314,7 @@ private MetaProperty establishStaticMetaProperty(MetaProperty mp) {
                 } else {
                     result = new MetaBeanProperty(propertyName, propertyType, getterMethod, setterMethod);
                 }
-            } else if (getter && !setter) {
+            } else if (getter /*&& !setter*/) {    // setter is unset
```

Comment:
I would remove the "// setter is unset" part. setter has been set, it just is implicitly always false, because we check for getter&&setter before going here in an elseif